scripts/workspace-conversion/workspace-to-dataset.py

```diff
@@ -0,0 +1,248 @@
+                wsAttributes.get("library:primaryDiseaseSite", None),
+                wsAttributes.get("library:studyDesign", None),
+                wsAttributes.get("library:cellType", None),
+            ],
+        )
+    )
+    metadata["counts"] = {}
+    metadata["counts"]["donors"] = wsAttributes.get("library:numSubjects", 0)
+    metadata["dct:dataCategory"] = wsAttributes.get("library:dataCategory", {}).get(
+        "items", None
+    )
+    metadata["TerraDCAT_ap:hasDataUsePermission"] = list(
+        filter(
+            None,
+            [
+                map_dataset_release_policy(
+                    wsAttributes.get("library:dataUseRestriction", "No restrictions")
+                )
+            ],
+        )
+    )
+    metadata["dct:title"] = wsAttributes.get("library:datasetName", None)
+    metadata["dct:version"] = wsAttributes.get("library:datasetVersion", None)
+    metadata["dct:description"] = wsAttributes.get("library:datasetDescription", None)
+    metadata["TerraDCAT_ap:hasOwner"] = wsAttributes.get("library:datasetOwner", None)
+    metadata["TerraDCAT_ap:hasDataCollection"] = []
+    if "library:datasetOwner" in wsAttributes:
+        metadata["TerraDCAT_ap:hasDataCollection"].append(
+            {"dct:identifier": wsAttributes["library:datasetOwner"]}
+        )
+        metadata["TerraDCAT_ap:hasOwner"] = wsAttributes["library:datasetOwner"]
+    metadata["TerraDCAT_ap:hasCustodian"] = wsAttributes["library:datasetCustodian"]
+    metadata["contributors"] = []
+    if (
+        "library:datasetDepositor" in wsAttributes
+        or "library:contactEmail" in wsAttributes
+    ):
+        contributor = {}
+        if "library:datasetDepositor" in wsAttributes:
+            contributor["contactName"] = wsAttributes["library:datasetDepositor"]
+            contributor["correspondingContributor"] = True
+        contributor["email"] = wsAttributes.get("library:contactEmail", None)
+        contributor["intstitution"] = next(
+            iter(wsAttributes.get("library:institute", {}).get("items", [])), None
+        )
+        metadata["contributors"].append(contributor)
+    metadata["prov:wasAssociatedWith"] = next(
+        iter(wsAttributes.get("library:institute", {}).get("items", [])), None
+    )
+    metadata["prov:wasGeneratedBy"] = []
+    if "library:projectName" in wsAttributes:
+        metadata["prov:wasGeneratedBy"].append(
+            {"TerraCore:hasAssayType": [wsAttributes["library:projectName"]]}
+        )
+    # Get the intersection of the datatype.items array and the data modality types
+    metadata["prov:wasGeneratedBy"].append(
+        {
+            "TerraCore:hasDataModality": map_data_modality(
+                wsAttributes.get("library:datatype", {}).get("items", [])
+            )
+        }
+    )
+    metadata["files"] = []
+    fileList = wsAttributes.get("library:dataFileFormats", {}).get("items", [])
+    for x in fileList:
```

Comment:
```suggestion
    for ext in fileList:
```

nit: minor suggestion for clarity